pkg/bitcoin/block.go

```diff
@@ -59,6 +62,39 @@ func (bh *BlockHeader) Serialize() [BlockHeaderByteLength]byte {
 	return result
 }
 
+// Deserialize deserializes a byte array to a BlockHeader using the block header
+// serialization format:
+// [Version][PreviousBlockHeaderHash][MerkleRootHash][Time][Bits][Nonce].
+func (bh *BlockHeader) Deserialize(rawBlockHeader [BlockHeaderByteLength]byte) {
+	offset := 0
+
+	// Version
+	bh.Version = int32(binary.LittleEndian.Uint32(rawBlockHeader[offset:]))
+	offset += 4
+
+	// PreviousBlockHeaderHash
+	copy(
+		bh.PreviousBlockHeaderHash[:],
+		rawBlockHeader[offset:offset+HashByteLength],
+	)
+	offset += HashByteLength
+
+	// MerkleRootHash
+	copy(bh.MerkleRootHash[:], rawBlockHeader[offset:offset+HashByteLength])
+	offset += HashByteLength
```

Comment:
Would it make sense to align the `Serialize` function to do `offset += HashByteLength` instead of `offset += len(bh.PreviousBlockHeaderHash)` and `offset += len(bh.MerkleRootHash)`? 

I like the consistency and if we are not sure that `HashByteLength` is future-proof then we need to think about modifying the `Serialize` function.

Reply:
It makes sense. Done in bfea8c6cebe61384628c03aef341e03843b49cae.